packages/client/machines/playlistMachine.ts

```diff
@@ -160,23 +163,59 @@ export type PlaylistMachineEvents = EventFrom<typeof playlistModel>;
 export type PlaylistActorRef = ActorRefFrom<PlaylistMachine>;
 
 interface CreatePlaylistMachineArgs {
-    initialState: MpeWorkflowState;
+    initialState: MpeWorkflowState | undefined;
+    roomID: string;
 }
 
 export function createPlaylistMachine({
     initialState,
+    roomID,
 }: CreatePlaylistMachineArgs): PlaylistMachine {
-    const roomID = initialState.roomID;
-
     return createMachine({
-        initial: 'idle',
+        initial: 'init',
 
         context: {
             ...playlistModel.initialContext,
-            state: initialState,
+            state: initialState || playlistModel.initialContext.state,
         },
 
         states: {
+            init: {
+                entry: send(playlistModel.events.GET_CONTEXT()),
+
+                on: {
+                    GET_CONTEXT: [
+                        {
+                            cond: () => {
+                                const playlistHasBeenSpawnedWithoutInitialContext =
+                                    initialState === undefined;
+                                return playlistHasBeenSpawnedWithoutInitialContext;
+                            },
+
+                            target: 'retrievingContext',
+                        },
+                        {
+                            target: 'idle',
+                        },
+                    ],
+                },
+            },
```

Comment:
It seems that what you wanted is an [_eventless transition_](https://xstate.js.org/docs/guides/transitions.html#eventless-always-transitions).

Reply:
Another pattern would to be dynamically compute the _initial_ state of the machine, depending on the parameter the machine creator received.